### PR TITLE
chore: Improve Dev Menu

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [] 

### Description

This PR makes the following changes
- Move items that are not used to the dev tools
- Add animation to the collapse menu to make it clear that it can be dismissed

https://user-images.githubusercontent.com/11945712/229542169-1d3d3c98-04ee-4a29-8ee9-caf2eb36dd62.mov

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by olerichter00 — APPROVED** [review]
🌟

File: src/app/utils/DevMenu.tsx [hunk 1]
```diff
@@ -258,6 +251,19 @@ export const DevMenu = ({ onClose = () => dismissModal() }: { onClose(): void })
               toast.show("Copied to clipboard", "middle")
             }}
           />
+          <FeatureFlagMenuItem
+            title="Open RN Dev Menu"
+            onPress={() => NativeModules.DevMenu.show()}
+          />
```
Comment (gkartalis) [inline src/app/utils/DevMenu.tsx:257]: **question(non-blocking):** Can we keep this up top? I think its really useful up there 🤩 especially when debugging but don't wanna block this (was also recently moved up there from @damassi)

**Review by dariakoko — DISMISSED** [review]
⭐

File: src/app/utils/DevMenu.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/utils/DevMenu.tsx:257]: ah cool - didn't know it was intentional. I haven't seen other folks using it. I can move it up of course

**Review by gkartalis — APPROVED** [review]
Thanks a lot! 🚀

**Pull request merged**